### DISCUSSION TITLE
go_router com flutter_modular, alguém já usou?

### DISCUSSION THREAD

**Discussion by Dreyfi** [Flutterando/forum#100] [Discussões gerais]
Fala devs!

Eu estou criando algumas melhorias no projeto da empresa e gostaria de trazer o flutter_modular, por já ter trabalhado com ele em diversos projetos anteriores. O ponto é que esse projeto atual é web, e pelo que pesquisei, o package go_router é muito recomendado pela comunidade pra solução de problemas relacionados a browsers entre outras coisas. Eu gostaria de usar todo o poder do modular deixando de lado apenas seu gerenciamento de rotas. O que vocês acham? Vai dar zerda? Obrigado!

**Comment by Zeca-dev** [comment]
Eu também já pensei em fazer isso, mas fiquei me perguntando como ficaria a inicialização dos módulos, já que são inicializados ao entrar nas rotas do modular.

**Comment by jacobaraujo7** [comment]
Os dois utilizam o Navigator 2.0, então terá que usar ou um ou outro.
Deixe-me saber o que lhe chama a atenção no go_router em comparação com o Modular.

**Reply by Zeca-dev** [reply]
No meu caso o motivo de usar o go_router seria a possibilidade de chamar rotas com o fullScreenDialog. Na empresa que trabalho tem muitos fluxos de tela onde se abre uma modal e segue o fluxo dentro dela, encerrado ao concluir ou ao fechar a modal. Tentei fazer isso com o modular e não consegui. Vi até que tem uma solicitação
desse recurso em aberto nas issues (https://github.com/Flutterando/modular/issues/730). 
Com o modular até tentei simular com a animação da rota com downToUp mas não ficou legal. Ao navegar com leftToRight não deu muito certo, pois aparecia um fundo preto na ida e na volta da rota. Eu gosto muito do modular mas esse recurso me faz falta.

**Comment by Dreyfi** [comment]
@jacobaraujo7 Eu estava com um problema no botão voltar do browser, chrome e safari. Em um lab que fiz utilizando go_router o problema parece ter sido resolvido. Mas após testar com Modular, vi que também funciona bem. Vou manter somente o Modular por enquanto. Muito obrigado pessoal.
